### PR TITLE
LWDEV-7940 - Change HFT service client to refit

### PR DESCRIPTION
**Pull request by njannink**
- Removed AutoRest client
- Added Refit client interfaces
- Moved models and duplicate objects to contracts

### CODE REVIEW

File: src/Lykke.Service.HFT.Contracts/History/LatestTradesQueryModel.cs [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    /// <summary>
+    /// Historic trades query model
+    /// </summary>
+    [PublicAPI]
+    public class LatestTradesQueryModel
+    {
+        /// <summary>
+        /// The trade asset ID.
+        /// </summary>
+        [Required]
```
Comment (polonzer) [inline src/Lykke.Service.HFT.Contracts/History/LatestTradesQueryModel.cs]: This parameter is not required.

File: src/Lykke.Service.HFT.Services/MatchingEngineAdapter.cs [hunk 2]
```diff
@@ -37,6 +37,11 @@ public class MatchingEngineAdapter : IMatchingEngineAdapter
             var response = await _matchingEngineClient.CancelLimitOrderAsync(limitOrderId.ToString());
             await CheckResponseAndThrowIfNull(response);
 
+            if (response.Status == MeStatusCodes.Ok)
```
Comment (polonzer) [inline src/Lykke.Service.HFT.Services/MatchingEngineAdapter.cs]: Why don't leave this at "ConvertToApiModel" method?

File: src/Lykke.Service.HFT/Controllers/OrderBooksController.cs [hunk 3]
```diff
@@ -58,7 +60,24 @@ public OrderBooksController(IOrderBooksService orderBooksService, IAssetServiceD
             }
 
             var orderBooks = await _orderBooksService.GetAsync(assetPairId);
-            return Ok(orderBooks);
+            return Ok(orderBooks.Select(ToModel));
+        }
+
+        private static OrderBookModel ToModel(OrderBook orderBook)
```
Comment (polonzer) [inline src/Lykke.Service.HFT/Controllers/OrderBooksController.cs]: This looks like wasting of CPU time. Can we return the same model here, without mapping?

File: src/Lykke.Service.HFT/Controllers/OrdersController.cs [hunk 4]
```diff
@@ -145,18 +144,19 @@ public class OrdersController : Controller
...
 
-            return Ok(order.ConvertToApiModel());
+            return Ok(ToModel(order));
         }
 
         /// <summary>
         /// Place a market order.
         /// </summary>
         /// <returns>Average strike price.</returns>
+        // TODO make this method obsolete and introduce a v2 that returns only double instead of ResponseModel{double}</remarks>
```
Comment (polonzer) [inline src/Lykke.Service.HFT/Controllers/OrdersController.cs]: We need to return proper DTO, not just doulbe.

File: src/Lykke.Service.HFT/Program.cs [hunk 5]
```diff
@@ -5,18 +5,18 @@
```
Comment (polonzer) [inline src/Lykke.Service.HFT/Program.cs:5]: Why do we need to change this file?

File: tests/Lykke.Service.HFT.Benchmarks/HftProductionBenchmark.cs [hunk 6]
```diff
@@ -1,101 +1,256 @@
...
     public class HftProductionBenchmark
     {
-        private const string HftApiUrl = @"http://localhost:5000";
-        private const string DevApiKey = @"<YOUR_TEST_API_KEY>";
-        private IHighFrequencytradingAPI _client;
-        private string[] _pairs;
-        private Random _randomPairIndex;
+        //private const string HftApiUrl = @"http://localhost:5000";
+        private const string HftApiUrl = @"https://hft-service-dev.lykkex.net/";
+        private const string DevApiKey = @"9b93e3ec-f9b0-42ba-b68c-cecb0d1e6dd4";
```
Comment (polonzer) [inline tests/Lykke.Service.HFT.Benchmarks/HftProductionBenchmark.cs]: Please never expose connection strings and passwords in source code.

File: src/Lykke.Service.HFT/Program.cs (hunk 5 above)
Comment (njannink) [reply src/Lykke.Service.HFT/Program.cs:5]: fixed resharper warnings

**Review by polonzer — APPROVED** [review]

**Pull request merged**